internal/api/v1alpha1/authproxyworkload_test.go

```diff
@@ -219,6 +220,135 @@ func TestAuthProxyWorkload_ValidateUpdate(t *testing.T) {
 					PortEnvName:      "DB_PORT",
 				}},
 			},
+		},
+		{
+			desc: "Valid when AuthProxyContainerSpec.RolloutStrategy goes from default to same explicit value",
+			spec: cloudsqlapi.AuthProxyWorkloadSpec{
+				Workload: cloudsqlapi.WorkloadSelectorSpec{
+					Kind: "Deployment",
+					Selector: &v1.LabelSelector{
+						MatchLabels: map[string]string{"app": "sample"},
+					},
+				},
+				Instances: []cloudsqlapi.InstanceSpec{{
+					ConnectionString: "proj:region:db2",
+					PortEnvName:      "DB_PORT",
+				}},
+			},
+			oldSpec: cloudsqlapi.AuthProxyWorkloadSpec{
+				Workload: cloudsqlapi.WorkloadSelectorSpec{
+					Kind: "Deployment",
+					Selector: &v1.LabelSelector{
+						MatchLabels: map[string]string{"app": "sample"},
+					},
+				},
+				AuthProxyContainer: &cloudsqlapi.AuthProxyContainerSpec{
+					RolloutStrategy: "Workload",
```

Comment:
This line is the key line right? The strategy is workload before, and not changed after?

Could we make that change more clear and hide the rest of these details?

Cf. https://arslan.io/2022/12/04/functional-table-driven-tests-in-go/

Reply:
Should we pull workload rollout strategy updates into a separate test?

Reply:
I moved RolloutStrategy update table tests into their own test function.